### PR TITLE
Dont use >= or <=

### PR DESCRIPTION
**Pull request by rhlsthrm**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol [hunk 1]
```diff
@@ -17,7 +17,7 @@ library Merkle {
 
     bytes32 proofElement;
     bytes32 computedHash = leaf;
-    for (uint256 i = 32; i <= proof.length; i += 32) {
+    for (uint256 i = 32; i < proof.length + 1; i += 32) {
```
Comment (Saw-mon-and-Natalie) [inline packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol:20]: `proof.length` is not a constant. This change does not necessarily save gas. 
- https://github.com/spearbit-audits/connext-nxtp/issues/98 does not apply.

**Comment by Saw-mon-and-Natalie** [comment]
@LayneHaber , the `2` new commits:

- https://github.com/connext/nxtp/pull/2514/commits/9a93cde4cbe86392f8472bd496f8d952850871ab
- https://github.com/connext/nxtp/pull/2514/commits/9a93cde4cbe86392f8472bd496f8d952850871ab

Do not follow the suggestion, since on one side of the inequality we would need a constant parameter, but in the above commits the side that has been updated is not a constant but a variable.

**Comment by LayneHaber** [comment]
reverted!

File: packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol (hunk 1 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol]: fixed: https://github.com/connext/nxtp/pull/2514/commits/46281c3ac81f3289ab7d94044f22f3b05580ae5b

File: packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol [hunk 2]
```diff
@@ -18,7 +18,7 @@ library Merkle {
     bytes32 proofElement;
     bytes32 computedHash = leaf;
     uint256 len = proof.length;
-    for (uint256 i = 32; i <= len; ) {
+    for (uint256 i = 32; i <= len + 1; ) {
```
Comment (gpersoon) [inline packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol:21]: I think this should be 
`for (uint256 i = 32; i < len + 1; ) { `
or
`for (uint256 i = 32; i <= len; ) {`

Comment (LayneHaber) [reply packages/deployments/contracts/contracts/messaging/connectors/polygon/lib/Merkle.sol:21]: fixed!

File: packages/deployments/contracts/contracts/core/connext/facets/SwapAdminFacet.sol [hunk 3]
```diff
@@ -157,7 +157,8 @@ contract SwapAdminFacet is BaseConnextFacet {
 
     // Check _pooledTokens and precisions parameter
     if (
-      _pooledTokens.length < Constants.MINIMUM_POOLED_TOKENS || _pooledTokens.length > Constants.MAXIMUM_POOLED_TOKENS
+      _pooledTokens.length < Constants.MINIMUM_POOLED_TOKENS ||
+      _pooledTokens.length > Constants.MAXIMUM_POOLED_TOKENS + 1
```
Comment (Saw-mon-and-Natalie) [inline packages/deployments/contracts/contracts/core/connext/facets/SwapAdminFacet.sol:161]: Needs to be:

```suggestion
      _pooledTokens.length > Constants.MAXIMUM_POOLED_TOKENS
```

**Comment by Saw-mon-and-Natalie** [comment]
We can apply the rule to:

https://github.com/connext/nxtp/blob/a7f314c65d317a83ff1d8281422a250a2971c317/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol#L1087

https://github.com/connext/nxtp/blob/a7f314c65d317a83ff1d8281422a250a2971c317/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol#L1100

**Comment by gpersoon** [comment]
Looks good, only latest suggestion by saw-mon and natalie doesn't seem to be included.


[nxtp/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol](https://github.com/connext/nxtp/blob/a7f314c65d317a83ff1d8281422a250a2971c317/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol#L1087)


`require(newAdminFee <= Constants.MAX_ADMIN_FEE, "too high"); `



[nxtp/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol](https://github.com/connext/nxtp/blob/a7f314c65d317a83ff1d8281422a250a2971c317/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol#L1100)



`require(newSwapFee <= Constants.MAX_SWAP_FEE, "too high");`

**Comment by LayneHaber** [comment]
tests failing due to offchain coverage

**Pull request merged**